File: pkg/codegen/testing/utils/testdata/typed-enum-pp/go/typed-enum.go
package main

import (
	storage "github.com/pulumi/pulumi-azure-native/sdk/go/azure/storage"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		someString := "foobar"
		typeVar := "Block"
		staticwebsite, err := storage.NewStorageAccountStaticWebsite(ctx, "staticwebsite", &storage.StorageAccountStaticWebsiteArgs{
			ResourceGroupName: pulumi.String(someString),
			AccountName:       pulumi.String(someString),
		})
		if err != nil {
			return err
		}
		_, err = storage.NewBlob(ctx, "faviconpng", &storage.BlobArgs{
			ResourceGroupName: pulumi.String(someString),
			AccountName:       pulumi.String(someString),
			ContainerName:     pulumi.String(someString),
			Type:              storage.BlobTypeBlock,
		})
		if err != nil {
			return err
		}
		_, err = storage.NewBlob(ctx, "_404html", &storage.BlobArgs{
			ResourceGroupName: pulumi.String(someString),
			AccountName:       pulumi.String(someString),
			ContainerName:     pulumi.String(someString),
			Type:              staticwebsite.IndexDocument.ApplyT(func(x *string) storage.BlobType { return storage.BlobType(*x) }).(storage.BlobTypeOutput),
		})
		if err != nil {
			return err
		}
		_, err = storage.NewBlob(ctx, "another", &storage.BlobArgs{
			ResourceGroupName: pulumi.String(someString),
			AccountName:       pulumi.String(someString),
			ContainerName:     pulumi.String(someString),
			Type:              storage.BlobType(typeVar),
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: pkg/codegen/testing/utils/testdata/typed-enum-pp/python/typed-enum.py
import pulumi
import pulumi_azure_native as azure_native

some_string = "foobar"
type_var = "Block"
staticwebsite = azure_native.storage.StorageAccountStaticWebsite("staticwebsite",
    resource_group_name=some_string,
    account_name=some_string)
# Safe enum
faviconpng = azure_native.storage.Blob("faviconpng",
    resource_group_name=some_string,
    account_name=some_string,
    container_name=some_string,
    type=azure_native.storage.BlobType.BLOCK)
# Output umsafe enum
_404html = azure_native.storage.Blob("_404html",
    resource_group_name=some_string,
    account_name=some_string,
    container_name=some_string,
    type=staticwebsite.index_document)
# Unsafe enum
another = azure_native.storage.Blob("another",
    resource_group_name=some_string,
    account_name=some_string,
    container_name=some_string,
    type=azure_native.storage.BlobType(type_var))
